### PR TITLE
Rename MyException to MOException

### PR DESCRIPTION
**Pull request by Holt59**
Rename `MyException` to `MOException`, keep an alias to avoid having to change all the projects right now. 

A few projects need updating because they inherit `MyException`, PR are coming (3 projects).

### CODE REVIEW

**Comment by isanae** [comment]
Why not just `Exception`? It's in `MOBase`. While you're at it, you could rename the namespace to `mo`...

**Comment by Holt59** [comment]
> 
> 
> Why not just `Exception`? It's in `MOBase`. While you're at it, you could rename the namespace to `mo`...

I can use `Exception`, I just want to get rid of `MyException`. I don't know if you're joking or not about `mo`...

**Comment by isanae** [comment]
I like `Exception` better. And no, I'm not joking. I'd rename `MOBase` and `MOShared` to `mo` and I'd put all of the modorganizer project in it too. But we'd probably want to discuss it first, so don't.

**Comment by Holt59** [comment]
> 
> 
> I like `Exception` better. And no, I'm not joking. I'd rename `MOBase` and `MOShared` to `mo` and I'd put all of the modorganizer project in it too. But we'd probably want to discuss it first, so don't.

Done.

**Comment by isanae** [comment]
Doesn't this break the ABI? I don't know what happens when an exported class suddenly becomes a typedef, but I don't think it works. I'd do it the other way around, or derive `Exception` from `MyException`.

**Comment by Holt59** [comment]
> 
> 
> Doesn't this break the ABI? I don't know what happens when an exported class suddenly becomes a typedef, but I don't think it works. I'd do it the other way around, or derive `Exception` from `MyException`.

What do you mean? It requires a re-compilation of all the projects but even without the name changes this is probably required due to the header change.

The problem with deriving is that you cannot catch them in the same way (e.g. if `Exception` derives from `MyException`, you cannot catch `Exception`, or the other way around).

**Comment by isanae** [comment]
I just mean that DLLs compiled against the old uibase won't load anymore. If that's not a problem, then sure. I thought we were trying to preserve the ABI, but I might have missed a bunch of discussions.

**Comment by Holt59** [comment]
> 
> 
> I just mean that DLLs compiled against the old uibase won't load anymore. If that's not a problem, then sure. I thought we were trying to preserve the ABI, but I might have missed a bunch of discussions.

The only DLL outside of the ModOrganizer2 organization I know is the Fallout76 one, but this one has been broken for a while anyway.

Also, there are a lots of PR to revert if we want this (the requirements, the `listSaves`, ...). I personally only try to maintain the Python interface when it's required.

File: src/imoinfo.h [hunk 1]
```diff
@@ -492,6 +493,10 @@ class QDLLEXPORT IOrganizer: public QObject
    */
   virtual bool onPluginDisabled(const QString& pluginName, std::function<void()> const& func) = 0;
 
+protected:
+
+  using QObject::QObject;
```
Comment (isanae) [inline src/imoinfo.h:498]: ?

Comment (Holt59) [reply src/imoinfo.h:498]: Oops, that should not be here 🙃

Comment (Holt59) [reply src/imoinfo.h:498]: That was to set the parent of `OrganizerProxy` in `modorganizer` but I'm using `setParent` instead.

**Comment by isanae** [comment]
All three look good.

**Review by isanae — APPROVED** [review]

**Pull request merged**